src/testdouble/cljs/csv.cljs

```diff
@@ -133,25 +132,29 @@
                              (conj rows []))))
 
                   :end-field
-                  (cond 
-                    (= str-char "\"")
-                    (recur (+ index 1) :in-field true field-buffer rows)
+                  (condp = str-char
+                    "\""
+                    (recur (inc index) :in-field true field-buffer rows)
 
-                    (= str-char separator)
-                    (recur (inc index) :end-field in-quoted-field "" (conj-in rows last-row-index ""))
+                    separator
+                    (recur (inc index)
+                           :end-field
+                           in-quoted-field
+                           ""
+                           (conj-in rows last-row-index ""))
 
-                    :else (recur (+ index 1) :in-field in-quoted-field str-char rows))
+                    (recur (inc index) :in-field in-quoted-field str-char rows))
 
                   :end-line
                   (case str-char
```

Comment:
could you use `condp` here too?

Reply:
Yep! I went with `case` to match https://guide.clojure.style/#case. I'm sure it's not the final word on how to write Clojure, but it's what I tend to refer back to when I don't know what I'm doing. 🙂